panic on artefactor publish when there are no files

```
$ mkdir -p test/downloads

$ cd test

$ ls downloads

$ artefactor --version
{Version:v0.3.6 Sha:42b2cfa80938cc93327d8b53b49f87482c23237b}

$ artefactor publish
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1184340]

goroutine 1 [running]:
github.com/appvia/artefactor/pkg/hashcache.GetFiles(0x1637636, 0x9, 0x0, 0x0, 0xc4201c5a00)
	/go/src/github.com/appvia/artefactor/pkg/hashcache/sha256.go:247 +0xc0
github.com/appvia/artefactor/pkg/docker.GetImages(0x1637636, 0x9, 0x0, 0x0, 0x0, 0x0, 0x16a7380, 0x5, 0x103e737)
	/go/src/github.com/appvia/artefactor/pkg/docker/docker.go:68 +0x6a
github.com/appvia/artefactor/pkg/cmd.publish(0x19ae080, 0x0, 0x0)
	/go/src/github.com/appvia/artefactor/pkg/cmd/publish.go:52 +0x21f
github.com/appvia/artefactor/pkg/cmd.glob..func4(0x19ae080, 0x19d2188, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/appvia/artefactor/pkg/cmd/publish.go:23 +0x2b
github.com/appvia/artefactor/vendor/github.com/spf13/cobra.(*Command).execute(0x19ae080, 0x19d2188, 0x0, 0x0, 0x19ae080, 0x19d2188)
	/go/src/github.com/appvia/artefactor/vendor/github.com/spf13/cobra/command.go:762 +0x468
github.com/appvia/artefactor/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x19aea00, 0x19aea00, 0xc4201c5f78, 0x1003abc)
	/go/src/github.com/appvia/artefactor/vendor/github.com/spf13/cobra/command.go:852 +0x30a
github.com/appvia/artefactor/vendor/github.com/spf13/cobra.(*Command).Execute(0x19aea00, 0x1, 0x0)
	/go/src/github.com/appvia/artefactor/vendor/github.com/spf13/cobra/command.go:800 +0x2b
github.com/appvia/artefactor/pkg/cmd.Execute()
	/go/src/github.com/appvia/artefactor/pkg/cmd/artefactor.go:109 +0x2d
main.main()
	/go/src/github.com/appvia/artefactor/cmd/artefactor/artefactor.go:21 +0x20
```

Just need to handle this error case more appropriately. When there are no files to publish, fail gracefully.

--- a/pkg/cmd/publish.go
+++ b/pkg/cmd/publish.go
@@ -59,6 +59,8 @@ func publish(c *cobra.Command) error {
 		if len(registry) < 1 {
 			return fmt.Errorf("must specify registry for publish")
 		}
+	} else {
+		fmt.Printf("No images to publish.\n")
 	}
 	for _, image := range images {
 		fmt.Printf("Loading image from %s.\n", image.FileName)

--- a/pkg/hashcache/sha256.go
+++ b/pkg/hashcache/sha256.go
@@ -76,12 +76,12 @@ func NewFromCheckSumsFile(file string, errIfMissing bool) (c *CheckSumCache, err
 		Dir:                 filepath.Dir(file),
 		CheckSumsByFilePath: make(map[string]CheckSumItem),
 	}
+	c.readCheckSumsIfPresent()
 	if errIfMissing {
 		if _, err := os.Stat(c.CheckSumFile); os.IsNotExist(err) {
-			return nil, fmt.Errorf("no checksum file %q", c.CheckSumFile)
+			return c, fmt.Errorf("no checksum file %q", c.CheckSumFile)
 		}
 	}
-	c.readCheckSumsIfPresent()
 	return c, nil
 }
 
@@ -180,6 +180,8 @@ func (c *CheckSumCache) Keep(file string) {
 
 // readCheckSumsIfPresent populates the hashcache from checksum file (if it exists)
 func (c *CheckSumCache) readCheckSumsIfPresent() {
+	// Re-init in memory checksums
+	c.CheckSumsByFilePath = make(map[string]CheckSumItem)
 	if _, err := os.Stat(c.CheckSumFile); err != nil {
 		// No checksums created... yet...
 		fullpath, _ := filepath.Abs(c.CheckSumFile)
@@ -191,9 +193,6 @@ func (c *CheckSumCache) readCheckSumsIfPresent() {
 		log.Fatal(err)
 	}
 	defer csf.Close()
-
-	// Re-init in memory checksums
-	c.CheckSumsByFilePath = make(map[string]CheckSumItem)
 	// open checksum file
 	scanner := bufio.NewScanner(csf)
 	for scanner.Scan() {

